presto-product-tests/src/main/java/io/prestosql/tests/hive/acid/TestPartitionedFullAcidTable.java

```diff
@@ -0,0 +1,134 @@
+        implements RequirementsProvider
+{
+    private static String tableName = "ORC_acid_single_int_column_partitioned_fullacid";
+
+    private static final HiveTableDefinition SINGLE_INT_COLUMN_PARTITIONED_ORC = singleIntColumnPartitionedTableDefinition(Optional.empty());
+
+    private static HiveTableDefinition singleIntColumnPartitionedTableDefinition(Optional<String> serde)
+    {
+        return HiveTableDefinition.builder(tableName)
+                .setCreateTableDDLTemplate(buildSingleIntColumnPartitionedTableDDL(serde))
+                .setNoData()
+                .build();
+    }
+
+    private static String buildSingleIntColumnPartitionedTableDDL(Optional<String> rowFormat)
+    {
+        StringBuilder sb = new StringBuilder();
+        sb.append("CREATE %EXTERNAL% TABLE IF NOT EXISTS %NAME%(");
+        sb.append("   col INT,");
+        sb.append("   fcol INT");
+        sb.append(") ");
+        sb.append("PARTITIONED BY (part_col INT) ");
+        if (rowFormat.isPresent()) {
+            sb.append("ROW FORMAT ").append(rowFormat.get());
+        }
+        sb.append(" STORED AS ORC");
+        sb.append(" TBLPROPERTIES ('transactional'='true')");
+        return sb.toString();
+    }
+
+    @Inject
+    private MutableTablesState tablesState;
+
+    @Override
+    public Requirement getRequirements(Configuration configuration)
+    {
+        return compose(mutableTable(SINGLE_INT_COLUMN_PARTITIONED_ORC));
+    }
+
+    @Test(groups = {HIVE_ACID, PROFILE_SPECIFIC_TESTS})
+    public void testReadingPartitionedInsertOnlyAcidTable()
+    {
+        String tableNameInDatabase = tablesState.get(tableName).getNameInDatabase();
+
+        QueryExecutor hiveQueryExecutor = ThreadLocalTestContextHolder.testContext().getDependency(QueryExecutor.class, "hive");
+        hiveQueryExecutor.executeQuery(
+                "INSERT OVERWRITE TABLE " + tableNameInDatabase + " PARTITION (part_col=2) select 21, 1");
+        hiveQueryExecutor.executeQuery(
+                "INSERT OVERWRITE TABLE " + tableNameInDatabase + " PARTITION (part_col=1) select 11, 1");
+
+        String selectFromOnePartitionsSql = "SELECT * FROM " + tableNameInDatabase + " WHERE part_col = 2";
+        QueryResult onePartitionQueryResult = query(selectFromOnePartitionsSql);
+        assertThat(onePartitionQueryResult).containsOnly(row(21, 1, 2));
+
+        hiveQueryExecutor.executeQuery(
+                "INSERT INTO TABLE " + tableNameInDatabase + " PARTITION (part_col=2) select 22, 2");
+        onePartitionQueryResult = query(selectFromOnePartitionsSql);
+        assertThat(onePartitionQueryResult).hasRowsCount(2);
+
+        // test filtering
+        onePartitionQueryResult = query(selectFromOnePartitionsSql + " AND fcol = 1");
+        assertThat(onePartitionQueryResult).containsOnly(row(21, 1, 2));
+
+        // delete a row
+        hiveQueryExecutor.executeQuery(
+                "DELETE FROM " + tableNameInDatabase + " where part_col=2 and fcol=2");
```

Comment:
`WHERE part_col = 2 AND fcol = 2` -- just as you did above in `" AND fcol = 1");`